syncer/syncer.go

```diff
@@ -281,13 +292,30 @@ func (s *Syncer) synchronize(ctx context.Context) bool {
 	// our clock starts ticking from 1 so it is safe to skip layer 0
 	// always sync to currentLayer-1 to reduce race with gossip and hare/tortoise
 	for layerID := s.mesh.ProcessedLayer().Add(1); layerID.Before(s.ticker.GetCurrentLayer()); layerID = layerID.Add(1) {
-		var layer *types.Layer
-		var err error
-		if layer, err = s.syncLayer(ctx, layerID); err != nil {
+		if layerID.FirstInEpoch() {
+			s.tbCount = make(map[string]int)
+			s.seenTB = make(map[string]struct{})
+			s.mostUsedTBCount = 0
+			s.mostUsedTB = nil
+		}
+
+		layer, err := s.syncLayer(ctx, layerID)
+		if err != nil {
 			logger.With().Error("failed to sync to layer", layerID, log.Err(err))
 			return false
 		}
 
+		if layerID.OrdinalInEpoch() < s.conf.TBCalculationLayers {
+			s.processTortoiseBeacons(ctx, layer)
+		} else {
+			epoch := layerID.GetEpoch()
```

Comment:
should this be done for each layer as of `s.conf.TBCalculationLayers` ?

Reply:
No, only one. Fixed it